docs/travis.md

```diff
@@ -0,0 +1,102 @@
+# Travis CI
+
+The Zulip server uses [Travis CI](https://travis-ci.org/) for its
+continuous integration.  This page documents useful tools and tips to
+know about when using Travis CI and debugging issues with it.
+
+## Goals
+
+The overall goal of our Travis CI setup is to catch as many possible
+future bugs in Zulip as possible, minimizing delay and false positives
+that waste the time of developers around the Zulip community
+debugging.  There are a few implications of this overall goal:
+
+* If a test is failing nondeterministically in Travis CI, we consider
+  that to be an urgent problem.
+* If the tests become a lot slower, that is also an urgent problem.
+* Everything we do in CI should be to the extent possible runnable in
+  an environment where one can iterate quickly, so that developers
+  only need to wait 10 minutes for Travis CI to run repeatedly when
+  modifying the Travis CI configuration itself.
+
+## Configuration
+
+The main Travis configuration file is `.travis.yml`.  The specific
+test suites we have are listed in the `matrix` section in that file.
+We've configured it to use a few helper scripts for each job:
+
+* `tools/travis/setup-$TEST_SUITE`: A Zulip script that sets up the
+  test environment (E.g. installing dependencies).
+  * For the backend and frontend suites, this is a thin wrapper around
+    `tools/provision`, aka the development environment provision script.
+  * For the production suite, this is a more complicated process
+    because of all the packages Travis installs.  See the comments in
+    `tools/travis/setup-production` for details.
+* `tools/travis/$TEST_SUITE`: A zulip script containing the actual
+  code for the test.
+
+The main purpose of the distinction between the two is that if the
+`setup-backend` job fails, Travis CI will report it as a build
+infrastructure failure, whereas if the `backend` job fails, it'll be
+reported as a test failure.
```

Comment:
They don't document it, it's just something you notice about the emails.  I'm going to leave it here, since I think it's useful higher-level explanation, though it could also be reasonable to add it to the `.travis.yml`